### PR TITLE
Use AtomicU64 now that it's stabilized

### PR DESCRIPTION
**Pull request by ryoqun**
Hi, guys!

This is my first and small pull-request.
Recently, I started to read solana's codebase.

This is still a draft; I left some comment to be removed after correctly understanding neighbor code.
I just wanted to run the CI.

If there is something wrong, please let me know.

Also, beware I'm new to both Rust and DLT... ;)

#### Problem

Use `AtomicU64` now that it's stabilized since Rust 1.34.0.
Because the size of `AtomicUsize` and `AtomicU64` is same for x86-64, there should be no risk for functional/runtime changes. 

Note that along with just renaming, I also removed a few assertions. That might affect the runtime in a negligible manner, however I assume `rustc` is clever enough to remove these constant expression assertions away. :) 

rust release note: https://github.com/rust-lang/rust/blob/master/RELEASES.md#version-1340-2019-04-11

#### Summary of Changes

Fixes a TODO comment in a source code.

### CODE REVIEW

File: runtime/src/bank.rs [hunk 1]
```diff
@@ -506,7 +506,7 @@ impl Bank {
         );
 
         self.capitalization.fetch_add(
-            (validator_rewards + storage_rewards) as usize,
+            (validator_rewards + storage_rewards) as u64, // WIP: why is it ok to cast f64 => u64?
```
Comment (mvines) [inline runtime/src/bank.rs]: We don't care about fractional lamports, I think that's why

**Comment by mvines** [comment]
Oh so CI probably failed because `cargo fmt` from 1.38 behaves differently than `cargo fmt` from 1.37.   If you upgrade to rust 1.38 and run `cargo fmt`, that should be cleared up.
Here's the exact location of the failure: https://buildkite.com/solana-labs/solana/builds/12585#cb474786-8345-43bb-a420-9b84bc613e9b/77

File: runtime/src/bank.rs (hunk 1 above)
Comment (rob-solana) [reply runtime/src/bank.rs]: yep

Comment (ryoqun) [reply runtime/src/bank.rs]: Thanks for the confirmation! I'm now sure that this is safe with my current understanding of the codebase too!

**Comment by ryoqun** [comment]
Thank for running CI. Seems I'm stumbled upon rather a preliminary step...
I polished my pull request up and flipped the draft flag!

**Comment by mvines** [comment]
@ryoqun - looks like clippy has a minor complaint now: https://buildkite.com/solana-labs/solana/builds/12612#640144ca-8b2b-42ba-9ac1-0c5cdf7e1e13/86-122 :)

**Comment by ryoqun** [comment]
@mvines I just fixed it. I wonder why it's complaining now. That unconditional cast was there before my pull request... clippy is getting smarter and/or clippy only checks touched lines?

**Comment by mvines** [comment]
Can you rebase your PR on the tip of the master branch please.  The CI run is picking up the older rust 1.37 version (we recently upgraded to rust 1.38), that's probably part of the CI unhappiness here

**Comment by mvines** [comment]
@ryoqun - I'm also happy to take this PR over from here if you'd prefer.  You're just running into random CI hiccups it seems 🎰.   Up to you :)

**Comment by ryoqun** [comment]
@mvines Sorry for being late.. I fixed another compilation issue and rebased my PR.

**Comment by mvines** [comment]
The coverage build failure is expected, rest looks great.    Thanks for your contribution @ryoqun!

**Comment by ryoqun** [comment]
@mvines Thanks for merging my first PR!

**Pull request merged**